core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -286,6 +286,9 @@ final void reprocess(@Nullable SubchannelPicker picker) {
 
     for (final PendingStream stream : toProcess) {
       PickResult pickResult = picker.pickSubchannel(stream.args);
+      if (pickResult.hasResult()) {
```

Comment:
This code is needed in newStream() as well. Let's only run it when `callOptions.isWaitForReady()` is true as well.